codec-redis/src/main/java/io/netty/handler/codec/redis/RedisDecoder.java

```diff
@@ -0,0 +1,253 @@
+        }
+        final long length = parseRedisNumber(lineByteBuf);
+        if (length < NULL_RESPONSE_LENGTH) {
+            throw new RedisCodecException("length: " + length + " (expected: >= -1)");
+        }
+        switch (type) {
+        case ARRAY_HEADER:
+            addMessageToOut(out, RedisMessageFactory.createArrayHeader(length));
+            return true;
+        case BULK_STRING:
+            if (length > REDIS_MESSAGE_MAX_LENGTH) {
+                throw new RedisCodecException("length: " + length + " (expected: <= " + REDIS_MESSAGE_MAX_LENGTH + ")");
+            }
+            bulkStringLength = (int) length; // range is already checked.
+            state = State.DECODE_BULK_STRING;
+            return true;
+        default:
+            throw new RedisCodecException("bad type: " + type);
+        }
+    }
+
+    private boolean decodeBulkString(ByteBuf in, List<Object> out) throws Exception {
+        switch (bulkStringLength) {
+        case NULL_RESPONSE_LENGTH:
+            // $-1\r\n
+            addMessageToOut(out, BulkStringRedisMessage.NULL_INSTANCE);
+            return true;
+        case 0:
+            // $0\r\n <here> \r\n
+            if (in.readableBytes() < CRLF_LENGTH) {
+                return false;
+            }
+            readEndOfLine(in);
+            addMessageToOut(out, BulkStringRedisMessage.EMPTY_INSTANCE);
+            return true;
+        default: // bulkStringLength is always positive.
+            // ${bulkStringLength}\r\n <here> {data...}\r\n
+            if (in.readableBytes() < bulkStringLength + CRLF_LENGTH) {
+                return false;
+            }
+            ByteBuf content = in.readSlice(bulkStringLength);
+            readEndOfLine(in);
+            addMessageToOut(out, RedisMessageFactory.createBulkString(content.retain()));
+            return true;
+        }
+    }
+
+    private static void readEndOfLine(final ByteBuf in) {
+        if (EOL == in.readShort()) {
+            return;
+        }
+        throw new RedisCodecException("expected: \\r\\n");
+    }
+
+    private void addMessageToOut(List<Object> out, RedisMessage msg) {
+        out.add(msg);
+        resetDecoder();
+    }
+
+    private RedisMessage newInlineRedisMessage(RedisMessageType messageType, ByteBuf bytes) {
+        switch (messageType) {
+        case SIMPLE_STRING:
+            return RedisMessageFactory.createSimpleString(bytes);
+        case ERROR:
+            return RedisMessageFactory.createError(bytes);
+        case INTEGER:
+            return RedisMessageFactory.createInteger(parseRedisNumber(bytes));
+        default:
+            throw new RedisCodecException("bad type: " + messageType);
+        }
+    }
+
+    private static ByteBuf readLine(ByteBuf in) {
+        final int lfIndex = in.forEachByte(ByteProcessor.FIND_LF);
+        if (lfIndex < 0) {
+            return null;
+        }
+        final int length = lfIndex - in.readerIndex() - 1; // `-1` is for CR
```

Comment:
actually it seems where this function is used `\n` and `\r` are not allowed in the content and so the `readEndOfLine` should catch any issues ... thanks!